src/make_watertight/CheckWatertight.cpp

```diff
@@ -0,0 +1,391 @@
+// *********************************
+
+// Patrick Shriwise
+// October, 2013
+
+// functions needed to check models for watertightness
+
+#include <iostream>
+#include <iomanip> // for setprecision
+#include <limits> // for min/max values
+#include <assert.h>
+#include <math.h>
+#include <time.h>
+#include <vector>
+#include <cmath>
+#include <cstdlib>
+#include <ctime>
+#include <set>
+#include <algorithm>
+
+// moab includes
+#include "moab/Core.hpp"
+#include "MBTagConventions.hpp"
+#include "moab/Range.hpp"
+#include "moab/Skinner.hpp"
+#include "moab/GeomTopoTool.hpp"
+
+#include "CheckWatertight.hpp"
+
+moab::ErrorCode CheckWatertight::check_mesh_for_watertightness(moab::EntityHandle input_set, double tol, bool& sealed, bool test, bool verbose, bool check_topology) {
+
+  moab::ErrorCode result;
+
+// create tags on geometry
+  moab::Tag geom_tag, id_tag;
+  result = MBI()->tag_get_handle(GEOM_DIMENSION_TAG_NAME, 1,
+                                 moab::MB_TYPE_INTEGER, geom_tag, moab::MB_TAG_DENSE | moab::MB_TAG_CREAT);
+  MB_CHK_SET_ERR(result, "could not get GEOM_DIMENSION_TAG_NAME handle");
+
+  result = MBI()->tag_get_handle(GLOBAL_ID_TAG_NAME, 1,
+                                 moab::MB_TYPE_INTEGER, id_tag, moab::MB_TAG_DENSE | moab::MB_TAG_CREAT);
+
+  MB_CHK_SET_ERR(result, "could not get GLOBAL_ID_TAG_NAME handle");
+
+
+  // get surface and volume sets
+  moab::Range surf_sets, vol_sets; // moab::Range of set of surfaces and volumes
+  // surface sets
+  int dim = 2;
+  void* input_dim[] = {&dim};
+  result = MBI()->get_entities_by_type_and_tag(input_set, moab::MBENTITYSET, &geom_tag,
+                                               input_dim, 1, surf_sets);
+  if (moab::MB_SUCCESS != result) {
+    return result;
+  }
+
+  // volume sets
+  dim = 3;
+  result = MBI()->get_entities_by_type_and_tag(input_set, moab::MBENTITYSET, &geom_tag,
+                                               input_dim, 1, vol_sets);
+  if (moab::MB_SUCCESS != result) {
+    return result;
+  }
+  if (!test) {
+    std::cout << "number of surfaces=" << surf_sets.size() << std::endl;
+    std::cout << "number of volumes="  << vol_sets.size() << std::endl;
+  }
+  // ******************************************************************
+  // Watertightness is a property of volumes. Check each surface for
+  // watertightness.
+  // ******************************************************************
+  // counted leaky surfaces
+  int total_counter = 0, unmatched_counter = 0;
+  std::set<int> leaky_surfs, leaky_vols;
+  moab::Skinner tool(MBI());
+
+  // remove all edges for fast skinning
+  moab::Range edges;
+
+  result = MBI()->get_entities_by_type(0, moab::MBEDGE, edges);   // get all edges
+  if (moab::MB_SUCCESS != result) { // failed to get edge data
+    return result; // failed
+  }
+//10/11/13
+//removed as a result of the change from the gen::find_skin function to the moab::Skinner:find_skin function
```

Comment:
Optional: If this has been removed / inactive since 2013, might make sense to delete it permanently.

Both here and throughout the file.